src/main/scala/uk/gov/nationalarchives/tdr/api/service/ClientFileMetadataService.scala

```diff
@@ -1,17 +1,26 @@
 package uk.gov.nationalarchives.tdr.api.service
 
-import java.sql.Timestamp
+import java.sql.{SQLException, Timestamp}
 import java.time.Instant
+import java.util.UUID
 
 import uk.gov.nationalarchives.Tables.ClientfilemetadataRow
 import uk.gov.nationalarchives.tdr.api.db.repository.ClientFileMetadataRepository
+import uk.gov.nationalarchives.tdr.api.graphql.DataExceptions.InputDataException
 import uk.gov.nationalarchives.tdr.api.graphql.fields.ClientFileMetadataFields.{AddClientFileMetadataInput, ClientFileMetadata}
 
 import scala.concurrent.{ExecutionContext, Future}
 
 class ClientFileMetadataService(clientFileMetadataRepository: ClientFileMetadataRepository, uuidSource: UUIDSource)
                                (implicit val executionContext: ExecutionContext) {
 
+  def getClientFileMetadata(fileId: UUID): Future[ClientFileMetadata] = {
+    clientFileMetadataRepository.getClientFileMetadata(fileId).map(_.head).map(rowToOutput).recover {
+      case _: NoSuchElementException => throw InputDataException(s"Could not find metadata for file $fileId")
+      case e: SQLException => throw InputDataException(e.getLocalizedMessage)
```

Comment:
Can you add the original exceptions as causes of the InputDataException so we don't lose the original stack trace?

Reply:
So the current case class looks like this :
`case class InputDataException(message: String) extends Exception(message)`
Do you mean change it to this and leave it at that?
`case class InputDataException(message: String, cause: Throwable) extends Exception(message, cause)`

The problem with this is that there is already a line in FileMetadataService which throws an exception without an original exception.
`case None => throw InputDataException(s"The property does not exist")` 
In this case, the property that has been passed in is wrong so this is very much an InputDataException.

And in terms of keeping the original stack trace, do we really need it here? The InputDataException will give the line number which threw it and from there we can easily work out what's going on without 100 lines of slick classes in the stack trace.

I'm open to adding it but at the moment it causes issues with the way it's being used elsewhere and doesn't seem to give us much benefit.

Reply:
If this was Java, you'd have an overloaded constructor. In Scala, I guess the best way to do it is with a default value:

```
class InputDataException(message: String, cause: Throwable = null) extends Exception(message, cause)
```

I'd always include the original exception to avoid throwing away information. It's hard to predict in advance what info will be useful when you're debugging something, and it's so easy to include an exception cause that my rule of thumb is to just do it.

Reply:
Fair point. I've changed it to be an option in the case class because null seems to be bad in Scala (https://docs.scala-lang.org/overviews/scala-book/no-null-values.html) although I have had to use .orNull because it's a java class.